src/shared/varlink.c

```diff
@@ -2299,10 +2324,16 @@ int varlink_server_listen_address(VarlinkServer *s, const char *address, mode_t
         if (listen(fd, SOMAXCONN) < 0)
                 return -errno;
 
-        r = varlink_server_listen_fd(s, fd);
+        r = varlink_server_create_listen_fd_socket(s, fd, &ss);
         if (r < 0)
                 return r;
 
+        addr_copy = strdup(address);
+        if (!addr_copy)
+                return log_oom_debug();
+
+        ss->address = TAKE_PTR(addr_copy);
```

Comment:
`free_and_strdup(&ss->address, address)`

Hmm, so looking at this part, this block open-codes `varlink_server_listen_fd()`. Why not do:
```
r = varlink_server_listen_fd(s, fd);
if (r < 0)
   return r;

r = free_and_strdup(&s->sockets->address, address);
if (r < 0)
   return log_oom_debug();
```
I'm not saying that's better, just wondering.

Reply:
I definitely considered it. But since `strdup()`/`free_and_strdup()` could fail, it felt like it went against the coding style if `varlink_server_listen_fd()` adds the `VarlinkServerSocket` object to the server's list if it failed to set the address.